test/provision_test.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+import json
+
+from autoprotocol.protocol import Protocol
+
+from test.test_util import TestUtils
+
+
+class TestProvision(object):
+    p = Protocol()
+    w1 = p.ref("w1", None, cont_type="96-pcr", discard=True) \
+        .well(0).set_volume("2:microliter")
+
+    def test_provision_well_capacity(self):
+        self.p.provision("rs17gmh5wafm5p", self.w1, "50:microliter")
+        actual_protocol_as_json = json.dumps(self.p.as_dict(), indent=2, sort_keys=True)
+        expected_protocol_as_json = TestUtils.read_json_file('expected_provision.json')
+
+        assert (expected_protocol_as_json == actual_protocol_as_json)
+
+    def test_attempt_to_provision_more_than_well_capacity(self):
+        with pytest.raises(ValueError):
+            self.p.provision("rs17gmh5wafm5p", self.w1, "500:microliter")
+
+    # TODO - If all container-types defined in `container_type.py` are of type Well, then code lines
+    # TODO - between 4960-4962 in protocol.py doesn't make any sense
+    # def test_attempt_to_provision_with_unsupported_container_type(self):
+    #     p = Protocol()
+    #     w1 = p.ref("w1", None, cont_type="micro-1.5", discard=True) \
+    #         .well(0).set_volume("2:microliter")
+    #     with pytest.raises(TypeError):
+    #         p.provision("rs17gmh5wafm5p", w1, "50:microliter")
```

Comment:
nit: remember to remove this before merging. Generally will prefer to surface these as issues rather than ephemerally in code, since the line numbers/context may change

Reply:
Intent is to bring the attention of core authors. Removed this code. Ideally I want to remove below code as well:
```
        if not is_valid_well(dests):
            raise TypeError("Destinations (dests) must be of type Well, "
                            "list of Wells, or WellGroup.")
```